qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/AbstractRestartUpgradeRestTestCase.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.neuralsearch.bwc;
+
+import java.util.Locale;
+import java.util.Optional;
+import org.junit.Before;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.neuralsearch.BaseNeuralSearchIT;
+import static org.opensearch.neuralsearch.TestUtils.CLIENT_TIMEOUT_VALUE;
+import static org.opensearch.neuralsearch.TestUtils.RESTART_UPGRADE_OLD_CLUSTER;
+import static org.opensearch.neuralsearch.TestUtils.BWC_VERSION;
+import static org.opensearch.neuralsearch.TestUtils.NEURAL_SEARCH_BWC_PREFIX;
+import org.opensearch.test.rest.OpenSearchRestTestCase;
+
+public abstract class AbstractRestartUpgradeRestTestCase extends BaseNeuralSearchIT {
+    protected static String testIndex;
+
+    @Before
+    protected void setIndex() {
```

Comment:
Could we just make this a getter function that tests can call - `getIndexNameForTest()`?

Reply:
I think there is no need of changing the index name here. Therefore before running the tests it will set the name and will increase the reusability of the variable here. Therefore if I make one extra method getIndexNameForTest() and call it will be addition of new code with a very less impact. I can easily use testIndex without creating extra object, without assigning any new value. Anyways I always delete the resources after the test is executed.

Reply:
In general, I prefer to avoid adding additional state to classes if its not needed. For this case, we could get rid of the testIndex variable and just have a single getter function name that can be called to get a test specific index name.